### PR TITLE
WIP:Ad/custom advancements

### PR DESCRIPTION
**Pull request by aidan99davis**

### CODE REVIEW

File: src/main/java/com/crowsofwar/avatar/common/event/ElementUnlockEvent.java [hunk 1]
```diff
@@ -10,6 +15,8 @@
 	public ElementUnlockEvent(EntityLivingBase entity, BendingStyle element) {
 		super(entity);
 		this.element = element;
+		AvatarLog.info("triggering element unlock");
+		AvatarTriggers.UNLOCK_ELEMENT.trigger((EntityPlayerMP) entity, element);
```
Comment (FavouriteDragon) [inline src/main/java/com/crowsofwar/avatar/common/event/ElementUnlockEvent.java:19]: Noooooooooo
Check to see if the entity is a player here, then trigger it. This event happens when abilities are used by npc's and mobs as well.

File: src/main/java/com/crowsofwar/avatar/common/triggers/AvatarTriggers.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.crowsofwar.avatar.common.triggers;
+
+import net.minecraft.advancements.CriteriaTriggers;
+import net.minecraft.advancements.ICriterionTrigger;
+
+public class AvatarTriggers {
+    public static final UnlockBendingTrigger UNLOCK_ELEMENT = new UnlockBendingTrigger("unlock_bending");
+    public static final UnlockAbilityTrigger ABILITY_USE = new UnlockAbilityTrigger("unlock_ability");
```
Comment (FavouriteDragon) [inline src/main/java/com/crowsofwar/avatar/common/triggers/AvatarTriggers.java:8]: Should be LevelAbilityTrigger- that way we can easily create custom advancements for unlocking it

File: src/main/java/com/crowsofwar/avatar/common/event/ElementUnlockEvent.java (hunk 1 above)
Comment (aidan99davis) [reply src/main/java/com/crowsofwar/avatar/common/event/ElementUnlockEvent.java:19]: image
Would this be enough to validate that? or is there a helper im missing

File: src/main/java/com/crowsofwar/avatar/common/event/AbilityLevelEvent.java [hunk 3]
```diff
@@ -15,6 +17,8 @@ public AbilityLevelEvent(EntityLivingBase entity, Ability ability, int oldLevel,
 		this.ability = ability;
 		this.oldLevel = oldLevel;
 		this.newLevel = newLevel;
+		if(entity instanceof EntityPlayerMP)
```
Comment (FavouriteDragon) [inline src/main/java/com/crowsofwar/avatar/common/event/AbilityLevelEvent.java:20]: Ok so technically this should go in an event hook with @SubscribeEvent, as we don't want this happening if the event is cancelled. However we shouldn't really run into that issue for now

File: src/main/java/com/crowsofwar/avatar/common/triggers/AvatarTriggers.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
```
Comment (FavouriteDragon) [inline src/main/java/com/crowsofwar/avatar/common/triggers/AvatarTriggers.java:8]: You need to fix your Ability_use trigger :p

**Pull request merged**